compiler/AST/build.cpp

```diff
@@ -378,11 +379,79 @@ static BlockStmt* buildUseList(BaseAST* module, BlockStmt* list) {
   }
 }
 
+//
+// Given an argument from a 'use' statement, process that argument
+// if it's a string (returning whether or not it was).  
+//
+// If it's a string literal, we assume it's either a "-llib" flag,
+// or a filename like "foo.h", "foo.c", "foo.o", etc.  
+//   - Peel off the former and pass them to addLibInfo(), the same
+//     function that handles command line -l flags.
+//   - Otherwise, assume it's the latter and pass it to our input
+//     file handler (which itself handles cases it doesn't recognize.
+//
+static bool processUseOfString(BaseAST* arg) {
+  //
+  // If we're 'use'ing a literal string...
+  //
+  if (SymExpr* se = toSymExpr(arg)) {
+    if (VarSymbol* var = toVarSymbol(se->var)) {
+      if (var->isImmediate()) {
+        Immediate* imm = var->immediate;
+        if (imm->const_kind == CONST_KIND_STRING) {
+          //
+          // And it starts with -l, treat it as a library specifier
+          //
+          if (strncmp(imm->v_string, "-l", 2) == 0) {
+            addLibInfo(astr(imm->v_string));
+          } else {
+            //
+            // otherwise, treat it as an input file
+            //
+            testInputFile(imm->v_string);
+          }
+          //
+          // In this case, we've consumed everything necessary, so
+          // just return the list to the callsite
+          //
+          return true;
+        }
+      }
+    }
+  }
+  return false;
+}
+
 
-BlockStmt* buildUseStmt(CallExpr* modules) {
+//
+// Build a 'use' statement
+//
+BlockStmt* buildUseStmt(CallExpr* args) {
   BlockStmt* list = NULL;
-  for_actuals(expr, modules)
-    list = buildUseList(expr->remove(), list);
+
+  //
+  // Iterate over the expressions being used, processing them
+  //
+  for_actuals(expr, args) {
+    Expr* useArg = expr->remove();
+    //
+    // if this is a string argument to 'use', process it
+    //
+    if (!processUseOfString(useArg)) {
+      //
+      // Otherwise, handle it in the traditional way
+      //
+      list = buildUseList(useArg, list);
+    }
+  }
+
+  //
+  // If all of them are consumed, replace the use statement by a no-op
+  //
+  if (list == NULL) {
+    list = new BlockStmt(new CallExpr(PRIM_NOOP));
```

Comment:
Might want to do

```c
list = buildChapelStmt(new CallExpr(PRIM_NOOP))
```

The only functional difference is that the BlockStmt built will be scopeless. [`buildChapelStmt` is what's used in `buildUseList`)]